pkg/app/AppService.go

```diff
@@ -280,154 +311,189 @@ func (impl AppServiceImpl) UpdateReleaseStatus(updateStatusRequest *bean.Release
+			isSucceeded, err = impl.UpdateDeploymentStatusForPipeline(app, pipelineOverride, cdWfr.Id)
 			if err != nil {
-				impl.logger.Errorw("error on update application status", "releaseCounter", releaseCounter, "gitHash", gitHash, "pipelineOverride", pipelineOverride, "dbApp", dbApp, "err", err)
-				return isHealthy, err
+				impl.logger.Errorw("error in updating deployment status for pipeline", "err", err)
+				return isSucceeded, isTimelineUpdated, err
 			}
-			if pipelineOverride.PipelineReleaseCounter == releaseCounter {
-				impl.logger.Debugw("inserting new app status", "status", string(newApp.Status.Health.Status), "dbApp", dbApp)
-				newDeploymentStatus := &repository.DeploymentStatus{
-					AppName:   newApp.Name,
-					AppId:     deploymentStatus.AppId,
-					EnvId:     deploymentStatus.EnvId,
-					Status:    string(newApp.Status.Health.Status),
-					CreatedOn: time.Now(),
-					UpdatedOn: time.Now(),
-				}
-				dbConnection := impl.pipelineRepository.GetConnection()
-				tx, err := dbConnection.Begin()
-				if err != nil {
-					impl.logger.Errorw("error on update status, db get txn failed", "CdWorkflowId", pipelineOverride.CdWorkflowId, "app", newApp, "err", err)
-					return isHealthy, err
-				}
-				// Rollback tx on error.
-				defer tx.Rollback()
-
-				err = impl.appListingRepository.SaveNewDeployment(newDeploymentStatus, tx)
-				if err != nil {
-					impl.logger.Errorw("error on saving new deployment status for wf", "CdWorkflowId", pipelineOverride.CdWorkflowId, "app", newApp, "err", err)
-					return isHealthy, err
-				}
-				err = impl.UpdateCdWorkflowRunnerByACDObject(newApp, pipelineOverride.CdWorkflowId)
-				if err != nil {
-					impl.logger.Errorw("error on update cd workflow runner", "CdWorkflowId", pipelineOverride.CdWorkflowId, "app", newApp, "err", err)
-					return isHealthy, err
-				}
-				err = tx.Commit()
-				if err != nil {
-					impl.logger.Errorw("error on db transaction commit for", "CdWorkflowId", pipelineOverride.CdWorkflowId, "app", newApp, "err", err)
-					return isHealthy, err
-				}
-				if string(application.Healthy) == newDeploymentStatus.Status {
-					isHealthy = true
-					impl.logger.Infow("writing cd success event", "gitHash", gitHash, "pipelineOverride", pipelineOverride, "newDeploymentStatus", newDeploymentStatus)
-					go impl.WriteCDSuccessEvent(newDeploymentStatus.AppId, newDeploymentStatus.EnvId, pipelineOverride)
-				}
-			} else {
-				impl.logger.Debug("event received for older triggered revision: " + gitHash)
+			if isSucceeded {
+				impl.logger.Infow("writing cd success event", "gitHash", gitHash, "pipelineOverride", pipelineOverride)
+				go impl.WriteCDSuccessEvent(cdPipeline.AppId, cdPipeline.EnvironmentId, pipelineOverride)
 			}
 		} else {
-			// new revision is not reconciled yet, thus status will not be changes and will remain in progress
+			impl.logger.Debug("event received for older triggered revision: " + gitHash)
```

Comment:
no string concatenation in logs. use key:val instead

Reply:
done